client/source/js/modules/project-set/program-set/program-modal.html

```diff
@@ -152,9 +152,9 @@ <h4 class="modal-title">{{ state.isNew ? 'Create' : 'Edit'}} program</h4>
               <div class="data numeric" style="margin-right: 15px"><i class="fa fa-trash-o" ng-click="deleteAddData(data)"></i></div>
             </div>
             <div class="column" ng-if="state.showAddData">
-              <div class="data"><input ng-model="state.newAddData.year"/></div>
-              <div class="data"><input ng-model="state.newAddData.spending"/></div>
-              <div class="data"><input ng-model="state.newAddData.coverage"/></div>
+              <div class="data"><input name="year" ng-model="state.newAddData.year" type="number" ng-class="{'invalid': programForm.year.$invalid}" min="{{1*state.program.addData[state.program.addData.length-1].year+1 || state.openProject.dataStart}}" max="{{state.openProject.dataEnd}}"/></div>
+              <div class="data"><input name="spending" ng-model="state.newAddData.spending" min="1" type="number" ng-class="{'invalid': programForm.spending.$invalid}"/></div>
```

Comment:
@guptaarun: I guess user may not add year data in ascending order always.
Since we do not allow editing, if user removed one record he will have to add again all of those. Can you rather validate that is year is not already added.